sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -662,7 +661,62 @@ Exceptions:
 
 |===
 
-Table 8. Member functions of the `command_graph` class (executable graph update).
+Table 8. Member functions of the `command_graph` class for queue recording.
+[cols="2a,a"]
+|===
+|Member function|Description
+
+|
+[source, c++]
+----
+using namespace ext::oneapi::experimental;
+bool begin_recording(const std::vector<queue>& queues)
+----
+
+|Synchronously changes the state of each queue in `queues` to the
+`queue_state::recording` state.
+
+Parameters:
+
+* `queues` - List of `sycl::queue` objects to change to the
+  `queue_state::recording` state and start recording commands to the graph
+  instance.
+
+Returns: `true` if any queue in `queues` has its state changed from
+`queue_state::executing` to `queue_state::recording`, `false` otherwise.
+
+Exceptions:
+
+* Throws synchronously with error code `invalid` if the any queue in `queues` is
+  already recording to a different graph.
```

Comment:
The reason we throw an exception in that case in our extension is that adding a new kernel to an existing fusion is more "dangerous" than adding a node to a graph. 

Because fusing kernels eliminates some of the implicit device-wide barriers/synchronization between individual kernels, in the case of kernel fusion there is a legality aspect to consider, i.e., will the resulting fused kernel produce correct results.

Therefore, we decided to throw an exception here, as the application obviously assumes that it is starting a new fusion, where in fact it isn't. 

IMHO, it is OK for the two extensions to diverge on this point, as adding a node to a graph is not nearly as "dangerous".